examples/src/kotlin/org/partiql/examples/MergeKeyValues.kt

```diff
@@ -0,0 +1,85 @@
+package org.partiql.examples
+
+import org.partiql.lang.eval.BindingCase
+import org.partiql.lang.eval.BindingName
+import org.partiql.lang.eval.EvaluationSession
+import org.partiql.lang.eval.ExprFunction
+import org.partiql.lang.eval.ExprValue
+import org.partiql.lang.eval.ExprValueFactory
+import org.partiql.lang.eval.ExprValueType
+import org.partiql.lang.eval.StructOrdering
+import org.partiql.lang.eval.namedValue
+import org.partiql.lang.eval.stringValue
+import org.partiql.lang.types.FunctionSignature
+import org.partiql.lang.types.StaticType
+import java.lang.Exception
+import kotlin.collections.HashMap
+
+abstract class MergeKeysBaseExprFunction(
+    val valueFactory: ExprValueFactory,
+) : ExprFunction
+
+/**
+ * For the Given [ExprValue] representing list of structs, merges key/values based on the given inputs in flatten list
+ * for values.
+ *
+ * E.g.
+ * Given:
+ *  [
+ *      {'Name':'certificate','Values':['abc', 'cde']},
+ *      {'Name':'certificate','Values':['ghj', 'klu']},
+ *      {'Name':'test','Values':['ghj', 'klu']}
+ *  ]
+ *  'Name' as mergeKey
+ *  'Values' as valueKey
+ *
+ *  Expected:
+ *  [
+ *      {'test': ['ghj', 'klu']},
+ *      {'certificate': ['abc', 'cde', 'ghj', 'klu']}
+ *  ]
+ */
+class MergeKeyValues(valueFactory: ExprValueFactory) :
+    MergeKeysBaseExprFunction(valueFactory) {
+    override val signature = FunctionSignature(
+        name = "merge_key_values",
+        requiredParameters = listOf(StaticType.BAG, StaticType.STRING, StaticType.STRING),
```

Comment:
The first required argument is a bag type, but the kdoc on L23 mentions a "list" and the example input uses a "list". Should the first required argument be changed to accept any collection type (i.e. bag, list, sexp)?

Reply:
I think it makes sense, let me look into this.